### PR TITLE
Support array type for evaluation

### LINKED ISSUE #17: Support array (Vec) type internally
The spec does not say anything about array in the [flag evaluation](https://openfeature.dev/specification/sections/flag-evaluation/) section.

However, some vendors may support array-like flag. In that case, when the provider implementer fetches JSON data from the server, there is no way to represent it using `evaluation::Value` type.

Implement it by:
* Adding a new variant to type `Value` to represent the type.

### PR DESCRIPTION
**Pull request by sheepduke**
Closes #17

### CODE REVIEW

**Comment by sheepduke** [comment]
Going to close this PR and just push the changes to unblock myself. Feel free to open an issue if you find anything not optimal. :-)

**Comment by sheepduke** [comment]
Seems the main branch is protected. Re-openning.

**Review by beeme1mr — APPROVED** [review]

**Comment by beeme1mr** [comment]
@sheepduke I'll be traveling this week but I'll try and help review when I can. Ping me if you get blocked by anything.

**Pull request merged**
